src/sentences/utils.rs

```diff
@@ -316,4 +316,70 @@ mod tests {
         let result = parse_magnetic_variation("12,Q");
         assert!(result.is_err());
     }
+
+    #[test]
+    fn test_parse_array_string() {
+        let result = array_string::<5>("12345");
+        assert!(result.is_ok());
+        let result = array_string::<5>("123456");
+        assert!(result.is_err());
+    }
+
+    #[test]
+    fn test_parse_number_in_range() {
+        let result = parse_number_in_range::<u8>("12", 10, 20);
+        assert!(result.is_ok());
+        let result = parse_number_in_range::<u8>("9", 10, 20);
+        assert!(result.is_err());
+        let result = parse_number_in_range::<u8>("21", 10, 20);
+        assert!(result.is_err());
+    }
+
+    #[test]
+    fn test_parse_number() {
+        let result = parse_num::<u8>("12");
+        assert!(result.is_ok());
+        let result = parse_num::<u8>("12.5");
+        assert!(result.is_err());
+    }
+
+    #[test]
+    fn test_parse_float_num() {
+        let result = parse_float_num::<f32>("12.5");
+        assert!(result.is_ok());
+        let result = parse_float_num::<f32>("12");
+        assert!(result.is_ok());
+    }
+
+    #[test]
+    fn test_parse_lat_lon() {
```

Comment:
To be 100% that test is failing for the reasons you assume, assert the Errors themselves.
you can even use something like `start_with` if, for example, you don't want to use the full string of the error.

Reply:
added explicit assertion for each error